### ISSUE TITLE
Json array and xml.toJson()

### ISSUE THREAD

**Issue by davebaol** [ballerina-platform/ballerina-lang#11794]
Well, I'm new to Ballerina but this sounds like a bug to me.

**Description:**
In case of an array the json generated by xml.toJson() is improperly recognized as map intead of json

**Steps to reproduce:**
The code below works as expected
````
public function main() {
    json person = {"person":{"name":"David"}};
    json persons = [person];
    persons[1] = person;
 
    io:println("persons: ", persons);
}
````
But if you replace the json literal assigned to `person` with the json generated by the xml conversion below you get the error `incompatible types: expected 'json', found 'map'`.
````
public function main() {
    xml xmlPerson = xml `<person><name>David</name></person>`;
    json person = xmlPerson.toJSON({});
    json persons = [person];       // <-- error
    persons[1] = person;           // <-- error
 
    io:println("persons: ", persons);
}
````
Not even casting person to `<json>`  makes it work. 😞 

**Affected Versions:**
Ballerina 0.983.0

**OS, DB, other environment details and versions:**
Windows 10

**Comment by pubudu91** [comment]
This was fixed by PR #12020 and will be available in the next release.

**Comment by davebaol** [comment]
Thanks!
